controllers/resource/resource_generator.go

```diff
@@ -369,3 +358,143 @@ func (r *resourceGenerator) CreateDoguSecret(doguResource *k8sv1.Dogu, stringDat
 func wrapControllerReferenceError(err error) error {
 	return fmt.Errorf("failed to set controller reference: %w", err)
 }
+
+type podSpecBuilder struct {
+	theDoguResource              *k8sv1.Dogu
+	theDogu                      *core.Dogu
+	metaAllLabels                k8sv1.CesMatchingLabels
+	specHostAliases              []corev1.HostAlias
+	specVolumes                  []corev1.Volume
+	specEnableServiceLinks       bool
+	specServiceAccountName       string
+	specInitContainers           []corev1.Container
+	specContainerCommand         []string
+	specContainerArgs            []string
+	specContainerLivenessProbe   *corev1.Probe
+	specContainerStartupProbe    *corev1.Probe
+	specContainerImagePullPolicy corev1.PullPolicy
+	specContainerVolumeMounts    []corev1.VolumeMount
+	specContainerEnvVars         []corev1.EnvVar
+	specContainerResourcesReq    corev1.ResourceRequirements
+}
+
+func newPodSpecBuilder(doguResource *k8sv1.Dogu, dogu *core.Dogu) *podSpecBuilder {
+	p := &podSpecBuilder{}
+	p.theDoguResource = doguResource
+	p.theDogu = dogu
+	return p
+}
+
+func (p *podSpecBuilder) labels(labels k8sv1.CesMatchingLabels) *podSpecBuilder {
+	p.metaAllLabels = labels
+	return p
+}
+
+func (p *podSpecBuilder) hostAliases(hostAliases []corev1.HostAlias) *podSpecBuilder {
+	p.specHostAliases = hostAliases
+	return p
+}
+
+func (p *podSpecBuilder) volumes(volumes []corev1.Volume) *podSpecBuilder {
+	p.specVolumes = volumes
+	return p
+}
+
+func (p *podSpecBuilder) enableServiceLinks(enable bool) *podSpecBuilder {
+	p.specEnableServiceLinks = enable
+	return p
+}
+
+func (p *podSpecBuilder) initContainers(initContainers ...*corev1.Container) *podSpecBuilder {
+	for _, initContainer := range initContainers {
+		if initContainer == nil {
+			continue
+		}
+
+		foundContainer := *initContainer
+		p.specInitContainers = append(p.specInitContainers, foundContainer)
+	}
+	return p
+}
+
+func (p *podSpecBuilder) containerCommandAndArgs(command, args []string) *podSpecBuilder {
+	p.specContainerCommand = command
+	p.specContainerArgs = args
+	return p
+}
+
+func (p *podSpecBuilder) containerLivenessProbe() *podSpecBuilder {
+	p.specContainerLivenessProbe = createLivenessProbe(p.theDogu)
+	return p
+}
+
+func (p *podSpecBuilder) containerStartupProbe() *podSpecBuilder {
+	p.specContainerStartupProbe = CreateStartupProbe(p.theDogu)
```

Comment:
add `startupProbeProbe` as Parameter instead of nested function-call

Reply:
i'd rather keep this one since the both methods `containerStartupProbe()` and `containerLivenessPrope()` work similar when viewed from the point of usage.